addons/hr_timesheet/models/project.py

```diff
@@ -81,18 +80,47 @@ def _search_is_internal_project(self, operator, value):
 
-    @api.depends('allow_timesheets', 'task_ids.planned_hours', 'task_ids.remaining_hours')
+    @api.model
+    def _fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        result = super()._fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
+        if view_type in ['tree', 'form'] and self.env.company.timesheet_encode_uom_id == self.env.ref('uom.product_uom_day'):
+            result['arch'] = self.env['account.analytic.line']._apply_time_label(result['arch'], related_model=self._name)
+        return result
+
+    def _get_uoms_and_timesheets_data(self):
+        timesheet_time_dict = defaultdict(list)
+        if not self:
+            return {}, timesheet_time_dict
+
+        timesheets_read_group = self.env['account.analytic.line'].read_group(
+            [('project_id', 'in', self.ids)],
+            ['project_id', 'unit_amount', 'product_uom_id'],
+            ['project_id', 'product_uom_id'],
+            lazy=False)
+        uom_ids = set(self.timesheet_encode_uom_id.ids)
+
+        for result in timesheets_read_group:
+            uom_id = result['product_uom_id'] and result['product_uom_id'][0]
+            if uom_id:
+                uom_ids.add(uom_id)
+            timesheet_time_dict[result['project_id'][0]].append((uom_id, result['unit_amount']))
+
+        uoms_dict = {uom.id: uom for uom in self.env['uom.uom'].browse(uom_ids)}
+
+        return uoms_dict, timesheet_time_dict
+
+    @api.depends('allow_timesheets', 'timesheet_ids')
     def _compute_remaining_hours(self):
-        group_read = self.env['project.task'].read_group(
-            domain=[('planned_hours', '!=', False), ('project_id', 'in', self.filtered('allow_timesheets').ids),
-                     '|', ('stage_id.fold', '=', False), ('stage_id', '=', False)],
-            fields=['planned_hours:sum', 'remaining_hours:sum'], groupby='project_id')
-        group_per_project_id = {group['project_id'][0]: group for group in group_read}
+        uoms_dict, timesheet_time_dict = self._get_uoms_and_timesheets_data()
         for project in self:
-            group = group_per_project_id.get(project.id, {})
-            project.remaining_hours = group.get('remaining_hours', 0)
-            project.has_planned_hours_tasks = bool(group.get('planned_hours', False))
-            project.is_project_overtime = group.get('remaining_hours', 0) < 0
+            total_time = sum([
+                unit_amount * uoms_dict.get(product_uom_id, project.timesheet_encode_uom_id).factor_inv
+                for product_uom_id, unit_amount in timesheet_time_dict[project.id]
+            ], 0.0)
+             # Now convert to the proper unit of measure set in the settings
+            total_time *= project.timesheet_encode_uom_id.factor
+            project.remaining_hours = project.allocated_hours - total_time
+            project.is_project_overtime = project.remaining_hours < 0
```

Comment:
I am getting confused now. Why do we want to convert? 
_compute_remaining_hours has to be in hours as its field name states, no?

Reply:
@IT-Ideas , 👍🏼  Agree with you. we do not need to convert the uom explicitly. it will be managed by the `timesheet_uom` widget. Thanks

Reply:
@IT-Ideas changes done